libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -410,7 +404,7 @@ internal class MapboxTripSession(
             launch {
```

Comment:
what is this needed for again?

Reply:
We had a ticket to investigate that https://github.com/mapbox/mapbox-navigation-android/issues/3265 (brought by https://github.com/mapbox/mapbox-navigation-android/pull/3255/files#r447617327 discussion).

From my testing it's needed to make sure that those don't block to each other - as they're executed in different dispatchers (`IO` and `Main`).

In any case, if you have time and can dig in deeper that'd be 💯 The more :eyes: the better 🍍